core/services/offchainreporting/data_source.go

```diff
@@ -23,8 +23,6 @@ type dataSource struct {
 var _ ocrtypes.DataSource = (*dataSource)(nil)
 
 // The context passed in here has a timeout of observationTimeout.
-// Gorm/pgx doesn't return a helpful error upon cancellation, so we manually check for cancellation and return a
-// appropriate error (FIXME: How does this work after the current refactoring?)
```

Comment:
Added a separate ctx for the tx where we insert the results, so we can save the results in the event of the observation ctx timing out.